--- includes/functions-auth.php.org	2013-04-19 12:16:23.000000000 +0200
+++ includes/functions-auth.php	2013-04-25 09:27:48.000000000 +0200
@@ -8,7 +8,15 @@
 	
 	if( $valid )
 		return true;
-		
+
+	// Hook for custom authentication plugins
+	if (yourls_has_action('user_auth')) {
+		if (yourls_do_action('user_auth'))
+		{
+			return $valid = true;
+		}
+	}
+
 	$unfiltered_valid = false;
 
 	// Logout request
